### PR TITLE
find_appcast: detect electron-builder appcasts

### LINKED ISSUE #69749: Detecting Electron appcasts
_Originally posted by @core-code in https://github.com/Homebrew/homebrew-cask/pull/69611#issuecomment-534687317_

i think i can add something to our current knowhow about finding appcasts for apps ( https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md )

many electron based apps contain a file in `Contents/Resources/app-update.yml `

this file contains the configuration how the app should be updates, i.e. the appcast information we are interested in

**example provider generic (http)**
```
provider: generic
url: 'https://desktop-release.notion-static.com/'
channel: latest
```

this means there will be a an appcast at URL+"latest-mac.yml"

**example provider s3**

```
provider: s3
bucket: timeular-desktop-packages
path: mac/production
```
this means there will be a an appcast at BUCKETNAME + ".s3.amazonaws.com/mac/production/latest-mac.yml"

other possible providers are "github" where the app is configured to use the github atom feed (which we can use too) or "bitbucket" which i haven't seen in practice yet

### PR DESCRIPTION
**Pull request by amyspark**
After making all changes to the cask:

- [ ] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [ ] The commit message includes the cask’s name and version.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [ ] Checked the cask is submitted to [the correct repo].
- [ ] Attempted to find an [appcast].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256
[appcast]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md

---

Fixes Homebrew/homebrew-cask#69749.

### CODE REVIEW

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
I’d rather not have a separate script just for this. We can either convert everything to ruby or convert the new script to bash (since keys are pretty consistent `key: value`, it should work).

I haven’t had the chance to look yet, but did you see if there are other providers? Some other apps to check are in https://github.com/Homebrew/homebrew-cask/pull/69611#issuecomment-535681740.

**Comment by amyspark** [comment]
@vitorgalvao I've made the changes you requested! As regards the Bash-vs-Ruby point: I know  Ruby can parse YAML with just two lines, that's why I stuck with it and called the helper from the main script.

File: developer/bin/find_appcast [hunk 1]
```diff
@@ -69,6 +69,18 @@ function find_devmate {
   verify_appcast 'DevMate' "${url}"
 }
 
+function find_electron_builder {
+  local app url
+
+  app="${1}"
+
+  electron_parser_script="$(brew --repo homebrew/cask)/developer/bin/parse_electron_builder_appcast"
```
Comment (vitorgalvao) [inline developer/bin/find_appcast]: Nice one! `brew --repo` is even better.

**Comment by core-code** [comment]
i've found another one that eludes the find_appcast search for an electron appcast:

https://s3.amazonaws.com/sm-app-updates/public/Smartmockups-2.7.3.dmg

**Comment by vitorgalvao** [comment]
@core-code Fixing in https://github.com/Homebrew/homebrew-cask/pull/71459.

**Pull request merged**